airflow/models/pool.py

```diff
@@ -64,6 +95,8 @@ def open_slots(self, session):
         from airflow.models.taskinstance import \
             TaskInstance as TI  # Avoid circular import
 
+        # Issue a single query instead of using the used_slots/queued_slots to
+        # avoid load on DB
         used_slots = session.query(func.count()).filter(TI.pool == self.pool).filter(
             TI.state.in_([State.RUNNING, State.QUEUED])).scalar()
         return self.slots - used_slots
```

Comment:
Before the [change removing this code](https://github.com/apache/airflow/commit/d63b6c9a5d3f8dfcb959daf93dd55dcb710d0e38), the open slots was calculated as slots - used_slots - queued_slots. Should we also restore it like that?

Reply:
used_slot already include both tasks in running state and queued state. So no need to minus the queued slots.

Reply:
I think it still is? used_slots was State.RUNNING, and queued_slots was State.QUEUED.

So now its `slots - (count RUNNING or QUEUED)` = `slots - RUNNING - QUEUED` (just via one query than two).

I think. It's been a long day.